packages/bitcore-node/src/routes/api/wallet.ts

```diff
@@ -134,6 +134,9 @@ router.get('/:pubKey/transactions', Auth.authenticateMiddleware, async (req: Aut
 
 router.get('/:pubKey/balance', Auth.authenticateMiddleware, async (req: AuthenticatedRequest, res) => {
   let { chain, network } = req.params;
+  if (req.body.tokenContractAddress) {
```

Comment:
I see why this line, but I think sending a body with a GET is probably bad practice

https://github.com/bitpay/bitcore/pull/2705/files#diff-ae61c7683311dfd83594b08ca457ae70R63
This method should not be sending a body, token address is meant to be a queryParam

client should have something like this in getBalance

```
if(tokenAddress) {
  url += `?tokenAddress=${tokenContractAddress}`
}
```
or

```
if(payload) {
  url += '?' + qs.stringify(payload)
}
```